IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Functions/IfcAlignmentRepresentation/Expression.txt

```diff
@@ -15,28 +17,39 @@
 	AlignmentRepresentations := Alignment.Representation.Representations;
 
 	(* one representation is either Curve2D or 3D *)
-	IF SIZEOF(AlignmentRepresentations) = 1
-	AND
-	(SIZEOF(QUERY(twoDee <* AlignmentRepresentations | twoDee.RepresentationType = 'Curve2D')) = 1
-	OR
-	SIZEOF(QUERY(threeDee <* AlignmentRepresentations | threeDee.RepresentationType = 'Curve3D')) = 1)
-	THEN
-		RETURN(TRUE);
+	IF SIZEOF(AlignmentRepresentations) = 1 THEN
+		IF SIZEOF(QUERY(twoDee <* AlignmentRepresentations | twoDee.RepresentationType = 'Curve2D')) = 1 THEN
+			CASE TRUE OF
+				'IFCGEOMETRYRESOURCE.IFCCOMPOSITECURVE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+				'IFCGEOMETRYRESOURCE.IFCOFFSETCURVEBYDISTANCE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+				'IFCGEOMETRYRESOURCE.IFCPOLYLINE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+		END_IF;
+		IF SIZEOF(QUERY(threeDee <* AlignmentRepresentations | threeDee.RepresentationType = 'Curve3D')) = 1
+				'IFCGEOMETRYRESOURCE.IFCGRADIENTCURVE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+				'IFCGEOMETRYRESOURCE.IFCOFFSETCURVEBYDISTANCE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+				'IFCGEOMETRYRESOURCE.IFCSEGMENTEDREFERENCECURVE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+				'IFCGEOMETRYRESOURCE.IFCPOLYLINE' IN TYPEOF(AlignmentRepresentations[1].Items[1]):
+					RETURN(TRUE);
+		END_IF;
 	END_IF;
 	
 	(* 2 representations and 3D BaseCurve reference equals 2D representation *)
 	IF SIZEOF(AlignmentRepresentations) = 2 THEN
 	
 		TwoDRep := QUERY(twoDee <* AlignmentRepresentations | twoDee.RepresentationType = 'Curve2D');
+		CurveTwoD := QUERY(item <* TwoDRep[1].Items | 'IFCGEOMETRYRESOURCE.IFCCOMPOSITECURVE' in TYPEOF(item));
 		ThreeDRep := QUERY(threeDee <* AlignmentRepresentations | threeDee.RepresentationType = 'Curve3D');
-		IF SIZEOF(ThreeDRep) = 1 AND ('IFCGEOMETRYRESOURCE.IFCGRADIENTCURVE' IN TYPEOF(ThreeDRep[1].Items[1])) THEN
-			IF SIZEOF(TwoDRep) = 1 AND ('IFCGEOMETRYRESOURCE.IFCCOMPOSITECURVE' IN TYPEOF(TwoDRep[1].Items[1])) THEN
-				IF ThreeDRep[1].Items[1]\IfcGradientCurve.BaseCurve :=: TwoDRep[1].Items[1] THEN
-					RETURN(TRUE);
-				END_IF;
-			END_IF;
-		END_IF;
+		CurveThreeD := QUERY(item <* ThreeDRep[1].Items | 'IFCGEOMETRYRESOURCE.IFCGRADIENTCURVE' in TYPEOF(item));
+		
+		RETURN(SIZEOF(CurveThreeD) = 1 AND SIZEOF(CurveTwoD) = 1 AND CurveThreeD\IfcGradientCurve.BaseCurve :=: CurveTwoD);
```

Comment:
Is it necessary to nominate `\IfcGradientCurve` to access `BaseCurve` attribute? The `CurveThreeD` variable is already of type `IfcGradientCurve`.

Reply:
I am just following suit from here: https://github.com/AECgeeks/infra-repo-issue-test-2/issues/16
since I did not get a definitive answer on my examples listed,

I did not understand the logic behind the change. It seems to me that p11 would see more importance on the IfcAxis2Placement example:
"NOTE This usage is required when a complex entity instance type has multiple attributes with the
same name or when a select data type contains multiple entities with attributes having the same name."

or even more important here:
```
LocationIs3D : SELF\IfcPlacement.Location.Dim = 3;
```
as Dim is not defined on IfcPoint

or here:
```
(Arg1.Dim <> Arg2.Dim)
```
where Arg1 and Arg2 are IfcVectorOrDirection where both entities have the same Dim attribute name.

I guess we are trusting someone more knowledgeable than us ->https://github.com/AECgeeks/infra-repo-issue-test-2/issues/16#issuecomment-1257575294 "but I trust the user of the tool to be highly knowledgeable."